src/reduxTypes.js

```diff
@@ -343,7 +343,7 @@ export type GlobalSettingsState = $ReadOnly<{
 
   // Possibly this should be per-account.  If so it should probably be put
   // on the server, so it can also be cross-device for the account.
-  doNotMarkMessagesAsRead: boolean,
+  doNotMarkMessagesAsRead: 'all' | 'stream' | 'topic' | 'never',
```

Comment:
Whenever changing a type that appears in the Redux state, we'll want a migration to update the user's existing data.

See discussion here:
https://github.com/zulip/zulip-mobile/blob/main/docs/howto/new-feature.md#redux-state-migrations